content/tracing/setup/docker.md

```diff
@@ -46,35 +46,85 @@ As with DogStatsD, traces can be submitted to the agent from other containers ei
 ### Using Docker links
 
 ```
-docker run  --name my_container           \
-            --all_your_flags              \
-            --link dd-agent:dd-agent    \
+docker run  --name my_container \
+            --all_your_flags \
+            --link dd-agent:dd-agent \
             my_image
 ```
 
-It expose `DD_AGENT_PORT_8126_TCP_ADDR` and `DD_AGENT_PORT_8126_TCP_PORT` as environment variables. Your application tracer can be configured to submit to this address.
+It expose `DD_AGENT_PORT_8126_TCP_ADDR` and `DD_AGENT_PORT_8126_TCP_PORT` as environment variables. Your application tracer must be configured to submit traces to this address.
 
-An example in Python:
+
+#### Python Configuration
 
 ```python
 import os
 from ddtrace import tracer
+
 tracer.configure(
-    hostname=os.environ["DD_AGENT_PORT_8126_TCP_ADDR"],
-    port=os.environ["DD_AGENT_PORT_8126_TCP_PORT"]
+    hostname=os.environ['DD_AGENT_PORT_8126_TCP_ADDR'],
+    port=os.environ['DD_AGENT_PORT_8126_TCP_PORT'],
+)
+```
+
+#### Ruby Configuration
+
+```ruby
+Datadog.configure do |c|
+  c.tracer hostname: ENV['DD_AGENT_PORT_8126_TCP_ADDR'],
+           port: ENV['DD_AGENT_PORT_8126_TCP_PORT']
+end
+```
+
+#### Go Configuration
+
+```go
+import (
+    "os"
+
+    ddtrace "github.com/DataDog/dd-trace-go/opentracing"
 )
+
+func main() {
+    config := ddtrace.NewConfiguration()
+    config.AgentHostname = os.GetEnv("DD_AGENT_PORT_8126_TCP_ADDR")
+    config.AgentPort = os.GetEnv("DD_AGENT_PORT_8126_TCP_PORT")
+
+    tracer, _, _ := ddtrace.NewTracer(config)
+}
+```
+
+#### Java Configuration
+
+You can update the Java Agent configuration via Environment Variables:
+
+```bash
+DD_AGENT_HOST=$DD_AGENT_PORT_8126_TCP_ADDR \
+DD_AGENT_PORT=$DD_AGENT_PORT_8126_TCP_PORT \
+java -javaagent:/path/to/the/dd-java-agent.jar -jar /your/app.jar
+```
+
+or via System Properties:
+
+```bash
+java -javaagent:/path/to/the/dd-java-agent.jar \
+     -Dagent.host=$DD_AGENT_PORT_8126_TCP_ADDR \
+     -Dagent.port=$DD_AGENT_PORT_8126_TCP_PORT \
+     -jar /your/app.jar
 ```
 
 ### Using Docker host IP
 
 Agent container port `8126` should be linked to the host directly. Having determined the address of the default route of this container, with `ip route` for example, you can configure your application tracer to report to it.
 
-An example in python, assuming `172.17.0.1` is the default route:
+The following is an example for the Python Tracer, assuming `172.17.0.1` is the default route:
 
-```
-from ddtrace import tracer; tracer.configure(hostname="172.17.0.1", port=8126)
+```python
+from ddtrace import tracer
+
+tracer.configure(hostname='172.17.0.1', port=8126)
```

Comment:
I think one example is enough here. We already did a configuration example for each language. Thoughts @lejulie?

Reply:
I think it's good to have example for each.

@l0k0ms do we have a widget where you can tab through the examples so they don't make the page very long?  Otherwise I'd still rather have the page be long and include more examples.  Let's not make anyone think more than they have to ;-).